ext/standard/tests/file/file_put_contents_5gb.phpt

```diff
@@ -0,0 +1,63 @@
+--TEST--
+Test file_put_contents() function with 5GB string
+--SKIPIF--
+<?php
+if (getenv('SKIP_SLOW_TESTS')) {
+    die('skip slow test');
+}
+
+function get_system_memory(): int|float|false
+{
+    if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
+        // Windows-based memory check
+        @exec('wmic OS get FreePhysicalMemory', $output);
+        if (isset($output[1])) {
+            return (int)trim($output[1]);
+        }
+    } else {
+        // Unix/Linux-based memory check
+        $memInfo = @file_get_contents("/proc/meminfo");
+        if ($memInfo) {
+            preg_match('/MemFree:\s+(\d+) kB/', $memInfo, $matches);
+            return $matches[1] * 1024; // Convert to bytes
+        }
+    }
+    return false;
+}
+if (get_system_memory() < 10 * 1024 * 1024 * 1024) {
+    die('skip Reason: Insufficient RAM (less than 10GB)');
+}
+$tmpfileh = tmpfile();
+if ($tmpfileh === false) {
+    die('skip Reason: Unable to create temporary file');
+}
+$tmpfile = stream_get_meta_data($tmpfileh)['uri'];
+if (disk_free_space(dirname($tmpfile)) < 10 * 1024 * 1024 * 1024) {
+    die('skip Reason: Insufficient disk space (less than 10GB)');
+}
+fclose($tmpfileh);
+
+?>
+
+--INI--
+memory_limit=6G
+
+--FILE--
+<?php
+
+$tmpfileh = tmpfile();
```

Comment:
I'm not sure that using tmpfile in test is a good idea. The problem is that if there is a crash before closing and after writing some data, the file might stay. We usually just create test file in the test dir and then use `--CLEAN--` section to remove.

Reply:
replaced it